erpnext/controllers/subcontracting_controller.py

```diff
@@ -109,6 +109,18 @@ def __identify_change_in_item_table(self):
 		self.__reference_name = []
 
 		if self.doctype in ["Purchase Order", "Subcontracting Order"] or self.is_new():
+			if self.doctype == "Subcontracting Order" and self.supplied_items:
+				self.sourced_by_supplier_items = []
+				for supplied_item in self.supplied_items:
+					if supplied_item.sourced_by_supplier:
+						self.sourced_by_supplier_items.append(
+							{
+								"rm_item_code": supplied_item.rm_item_code,
+								"rate": supplied_item.rate,
+								"amount": supplied_item.amount,
+								"sourced_by_supplier": supplied_item.sourced_by_supplier,
+							}
+						)
```

Comment:
Is this a child table? Or a random class variable?

If it's the first one, you probably wanna do:

```suggestion
						self.append(
							"sourced_by_supplier_items",
							{
								"rm_item_code": supplied_item.rm_item_code,
								"rate": supplied_item.rate,
								"amount": supplied_item.amount,
								"sourced_by_supplier": supplied_item.sourced_by_supplier,
							}
						)
```

That will use Frappe's ORM, and probably avoid the issues that is making you do `self.get("sourced_by_supplier_items")` below

Reply:
I am using `sourced_by_supplier_items` random class variable here, `supplied_items` is the child table